src/components/getApplicationData.ts

```diff
@@ -1,27 +1,30 @@
-import { ActionPayload } from '../types'
+import { ActionApplicationData, ActionPayload } from '../types'
 import DBConnect from './databaseConnect'
 import { BasicObject } from '@openmsupply/expression-evaluator/lib/types'
 import { getAppEntryPointDir } from './utilityFunctions'
 import config from '../config'
 
 // Add more data (such as org/review, etc.) here as required
-export const getApplicationData = async (payload: ActionPayload) => {
-  const { trigger_payload } = payload
-  if (!trigger_payload) throw new Error('trigger_payload required')
-  const applicationId = await DBConnect.getApplicationIdFromTrigger(
-    trigger_payload.table,
-    trigger_payload.record_id
-  )
+export const getApplicationData = async (input: {
+  payload?: ActionPayload
+  applicationId?: number
+  reviewId?: number
+}): Promise<ActionApplicationData> => {
+  // Requires either application OR trigger_payload, so throw error if neither provided
+  if (!input?.payload?.trigger_payload && !input?.applicationId)
+    throw new Error('trigger_payload or applicationId required')
+
+  const { trigger_payload } = input?.payload as ActionPayload
```

Comment:
If you would receive only an applicationId would it be a problem here?

Reply:
Also I don't. see the case where only `reviewId` is received... Would in both cases `applicationId` and `reviewId` one payload gets passed?

Reply:
> If you would receive only an applicationId would it be a problem here?

No, it's fine. That's what this change does -- it means we can generate `applicationData` from just an application Id rather than needing a whole Action payload.

> Also I don't. see the case where only `reviewId` is received... Would in both cases `applicationId` and `reviewId` one payload gets passed?

You can't pass only `reviewId` -- it's "extra" on top of basic application data -- it would cause problems to generate only the review part of this without the relevant application info.

So it throws an error if only `reviewId` is supplied.